### DISCUSSION TITLE
Considering Updating ECMAScript Version of Rule Engine Script Node

### DISCUSSION THREAD

**Discussion by kuraykaraaslan** [thingsboard/thingsboard#11558] [Ideas]
**Hello ThingsBoard Community,**

I was thinking it would be great to update the ECMAScript (JavaScript) version of Rule Engine Script Node. By switching to a newer version, we can use modern language features, including better support for **async** functions. 

**This will help us write cleaner, more efficient code, especially when dealing with asynchronous operations.**

Current version doesn't support a lot of features like async

**Comment by SemaGoktas** [comment]
Yes, this will definitely be useful. It will provide significant assistance and make rules at hand easier to manage. Overall, it will contribute positively and help us achieve our objectives more efficiently.

**Comment by JosAndrae** [comment]
why is every answer sounding like a ChatGPT text
